File: Mooga/src/de/lumpn/mooga/Individual.java
package de.lumpn.mooga;

public interface Individual {

	/**
	 * Returns the individual's genome.
	 */
	public Genome getGenome();

	/**
	 * Number of attributes evaluated.
	 */
	public int numAttributes();
	
	/**
	 * Priority of specific attribute for multi stage multi object optimization
	 */
	public int getPriority(int attribute);

	/**
	 * Score of specific attribute. Make sure getting these values is fast (use cache)!
	 */
	public double getScore(int attribute);

	/**
	 * Wether or not this individual can be considered best
	 */
	public boolean isElite();

	@Override
	public int hashCode();

	@Override
	public boolean equals(Object obj);
}

File: csharp/InfiniteZelda/Mooga/Individual.cs
namespace Lumpn.Mooga
{
    public interface Individual
    {
        /// The individual's genome.
        Genome Genome { get; }

        // TODO Jonas: numAttributes is not an attribute of individuals but of the configuration
        /// Number of attributes evaluated.
        int NumAttributes { get; }

        // TODO Jonas: priority is not an attribute of individuals but of the configuration
        /// Priority of specific attribute for multi stage multi object optimization
        int Priority(int attribute);

        /// Score of specific attribute. Make sure getting these values is fast (use cache)!
        double Score(int attribute);

        // TODO Jonas: remove
        /// Whether or not this individual can be considered best
        bool IsElite { get; }
    }
}
